pkg/status/status_test.go

```diff
@@ -0,0 +1,785 @@
+			}},
+		},
+	}
+)
+
+func TestBackends(t *testing.T) {
+	var backends Backends
+
+	backends.AddURL("external", url.URL{Host: "www.example.com"})
+	backends.AddURL("cluster", url.URL{Host: "www.example.com"})
+	backends.AddURL("cluster", url.URL{Host: "www.blah.com"})
+
+	expected := map[Visibility]URLSet{
+		"external": sets.New(
+			url.URL{Host: "www.example.com"},
+		),
+		"cluster": sets.New(
+			url.URL{Host: "www.example.com"},
+			url.URL{Host: "www.blah.com"},
+		),
+	}
+
+	if diff := cmp.Diff(expected, backends.URLs); diff != "" {
+		t.Error("unexpected diff (-want,+got): ", diff)
+	}
+}
+
+func TestProbeAllHosts(t *testing.T) {
+	const hostA = "foo.bar.com"
+	const hostB = "ksvc.test.dev"
+	var hostBEnabled atomic.Bool
+
+	hash := "some-hash"
+
+	// Failing handler returning HTTP 500 (it should never be called during probing)
+	failedRequests := make(chan *http.Request)
+	failHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		failedRequests <- r
+		w.WriteHeader(500)
+	})
+
+	// Actual probe handler used in Activator and Queue-Proxy
+	probeHandler := probe.NewHandler(failHandler)
+
+	// Probes to hostA always succeed and probes to hostB only succeed if hostBEnabled is true
+	probeRequests := make(chan *http.Request)
+	finalHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		probeRequests <- r
+		if !strings.HasPrefix(r.Host, hostA) &&
+			(!hostBEnabled.Load() || !strings.HasPrefix(r.Host, hostB)) {
+			w.WriteHeader(http.StatusNotFound)
+			return
+		}
+
+		r.Header.Set(header.HashKey, hash)
+		probeHandler.ServeHTTP(w, r)
+	})
+
+	ts := httptest.NewServer(finalHandler)
+	defer ts.Close()
+	tsURL, err := url.Parse(ts.URL)
+	if err != nil {
+		t.Fatalf("Failed to parse URL %q: %v", ts.URL, err)
+	}
+
+	hostAURL := *tsURL
+	hostAURL.Host = hostA
+	hostBURL := *tsURL
+	hostBURL.Host = hostB
+
+	ready := make(chan types.NamespacedName)
+	prober := NewProber(
+		zaptest.NewLogger(t).Sugar(),
+		fakeProbeTargetLister{
+			PodIPs:  sets.New(tsURL.Hostname()),
+			PodPort: tsURL.Port(),
+		},
+		func(ing types.NamespacedName) {
+			ready <- ing
+		})
+
+	done := make(chan struct{})
+	cancelled := prober.Start(done)
+	defer func() {
+		close(done)
+		<-cancelled
+	}()
+
+	// The first call to IsReady must succeed and return false
+	backends := Backends{
+		Key:     ingressNN,
+		Version: hash,
+		URLs: map[v1alpha1.IngressVisibility]URLSet{
+			v1alpha1.IngressVisibilityExternalIP: sets.New(
+				hostAURL, hostBURL,
+			),
+		},
+	}
+
+	state, active := prober.IsProbeActive(ingressNN)
+	if active {
+		t.Error("probe as active when it should have been false")
+	}
+	if diff := cmp.Diff(ProbeState{}, state); diff != "" {
+		t.Error("inactive probe should have empty state: ", diff)
+	}
+
+	pstate, err := prober.DoProbes(context.Background(), backends)
+	if err != nil {
+		t.Fatal("IsReady failed:", err)
+	}
+	if pstate.Ready == true {
+		t.Fatal("IsReady() returned true")
```

Comment:
I noticed multiple messages in the test file using "IsReady()" text, that could be modified a little. Unless you are already planning to do that in subsequent PRs.

Reply:
+1, also the error message would be misleading (was before as well).

Reply:
done